### DISCUSSION TITLE
Customize the search box or change the refiners to being string with contains

### DISCUSSION THREAD

**Discussion by rachelmarshall** [microsoft-search/pnp-modern-search#3009] [Ideas]
**Is your feature request related to a problem? Please describe. **

It would be helpful to search on specific fields like first name or last name.  If you could customize the search box to filter on these search properties with contains or beings with.  If the refiners could be a string with contains or being with.

For example, I would like to have an employee directory and to search on location, list name, last name, and or job title.

**Additional context**
![image](https://user-images.githubusercontent.com/18274244/109586860-0de99780-7ad4-11eb-9e6a-3f97f7538138.png)
![image](https://user-images.githubusercontent.com/18274244/109586909-26f24880-7ad4-11eb-9c63-01079c141c3d.png)

**Comment by wobba** [comment]
You can set up search to pull people and use the filter web part for JobTitle,Location etc. For firstname and lastname you have to create your own refinable managed properties as there are not refinable oob. Using the combo-box layout you can filter on the values as well.

If you want something customized you can always create your own filter type web part, and have it work thru dynamic data with the result web part.

**Comment by rachelmarshall** [comment]
I have refinable managed properties.  The issue in the filter/ refinable web part you only have the option of the combo box.  I need a string starting with the option as an operator.

I am very sorry I have put my comments here.

**Comment by wobba** [comment]
I'll tag this with an enhancement for the v4 parts to allow contains filtering.

**Comment by PathToSharePoint** [comment]
@wobba Did you start anything on this?

I was thinking it would be a good one for an employee directory (and that's actually how I found this thread). Refiners sound a bit complex to me, so I was thinking about playing with the searchbox:
- pass contains query instead of equals in the connection
- add radio buttons to search by job title, department, etc.

For example, selecting the Department radio button and typing `comm `would send the query `Department:comm*`

As for the refiners, I have a side question: is it expected that Department is not available out of the box? Does it mean we need to create custom managed properties just like first name and last name? How about the other fields, like work phone or office location?
